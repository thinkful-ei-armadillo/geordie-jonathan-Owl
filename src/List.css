img {
  width: 50px;
  height: auto;
  border-radius: 100%;
}

.name {
  display: inline-block;
  font-size: 16px;
}
