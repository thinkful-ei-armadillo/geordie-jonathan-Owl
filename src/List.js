import React from 'react';
import './List.css';
// generate participant name, avatar, and inSession(inSession should be shown before left session)

function List(props) {
  const badge = +props.unread
  ? <div className="unread_count badge">{props.unread}</div>
  : null;
  
  return (
    <div>
      
      <img src={props.avatar} alt={(props.name + ' avatar')}></img>
      <h2 className="name">{props.name}</h2>
      <p>{props.inSession}</p>
      <p>{badge}</p>
    </div>
  )
}

export default List;
