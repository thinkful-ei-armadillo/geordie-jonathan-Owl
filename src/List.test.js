import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

const listArr = [{
  name: 'name',
  avatar: 'avatar',
  inSession: true
}];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List listArr={listArr} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders expected list', () => {
  const tree = renderer
    .create(<List people={listArr} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
