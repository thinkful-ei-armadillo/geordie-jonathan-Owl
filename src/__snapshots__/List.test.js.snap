// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expected list 1`] = `
<div>
  <img
    alt="avatar"
  />
  <h2
    className="name"
  />
  <p />
  <p />
</div>
`;
